controllers/groceries.js

```diff
@@ -0,0 +1,108 @@
+const Groceries = require('../models/Groceries.js')
+
+const getAllGroceries = async (req, res) => {
+    try {
+        const groceries = await Groceries.find({ createdBy: req.user.id }).sort("createdAt");
+        res.render("groceries", { groceries });
+    } catch (error) {
+        res.status(500).json({error})
```

Comment:
If some error happen here, you need to add an error to the req.flash and redirect the user to the index page.
Ex:
```
} catch (error) {
        req.flash('error', 'Undefined error')
        res.redirect("/"); 
    }
```